From f4492fb683013d7de6f4788df6deb24fed8671b4 Mon Sep 17 00:00:00 2001
From: Nick Clifton <nickc@redhat.com>
Date: Thu, 1 Sep 2022 14:51:50 +0100
Subject: [PATCH] dllwrap, windres and dlltools use mktemp, which should be
 avoided

	PR 29534
	* dllwrap.c: Replace uses of choose_temp_base() with
	make_temp_file().
	* dlltool.c: Likewise.
	* resrc.c: Likewise.
    
Note: hunk for changelog has been removed to avoid conflicts    
---
 binutils/dlltool.c | 2 +-
 binutils/dllwrap.c | 6 +++---
 binutils/resrc.c   | 4 ++--
 4 files changed, 14 insertions(+), 6 deletions(-)

diff --git a/binutils/dlltool.c b/binutils/dlltool.c
index b4ac842212a..a3c5e0f778e 100644
--- a/binutils/dlltool.c
+++ b/binutils/dlltool.c
@@ -1270,7 +1270,7 @@ run (const char *what, char *args)
   int i;
   const char **argv;
   char *errmsg_fmt = NULL, *errmsg_arg = NULL;
-  char *temp_base = choose_temp_base ();
+  char *temp_base = make_temp_file ("");
 
   inform (_("run: %s %s"), what, args);
 
diff --git a/binutils/dllwrap.c b/binutils/dllwrap.c
index 1be3cec72f1..7c77701f4e9 100644
--- a/binutils/dllwrap.c
+++ b/binutils/dllwrap.c
@@ -351,7 +351,7 @@ run (const char *what, char *args)
   int i;
   const char **argv;
   char *errmsg_fmt = NULL, *errmsg_arg = NULL;
-  char *temp_base = choose_temp_base ();
+  char *temp_base = make_temp_file (NULL);
   int in_quote;
   char sep;
 
@@ -818,7 +818,7 @@ main (int argc, char **argv)
 
   if (! def_file_seen)
     {
-      char *fileprefix = choose_temp_base ();
+      char *fileprefix = make_temp_file (NULL);
 
       def_file_name = (char *) xmalloc (strlen (fileprefix) + 5);
       sprintf (def_file_name, "%s.def",
@@ -1031,7 +1031,7 @@ Creating one, but that may not be what you want"));
 
   if (! base_file_name)
     {
-      char *fileprefix = choose_temp_base ();
+      char *fileprefix = make_temp_file (NULL);
       base_file_name = (char *) xmalloc (strlen (fileprefix) + 6);
       sprintf (base_file_name, "%s.base",
 	       (dontdeltemps) ? mybasename (fileprefix) : fileprefix);
diff --git a/binutils/resrc.c b/binutils/resrc.c
index fff77f4f28a..2cf168e0f58 100644
--- a/binutils/resrc.c
+++ b/binutils/resrc.c
@@ -202,7 +202,7 @@ run_cmd (char *cmd, const char *redir)
   int i;
   const char **argv;
   char *errmsg_fmt = NULL, *errmsg_arg = NULL;
-  char *temp_base = choose_temp_base ();
+  char *temp_base = make_temp_file (NULL);
   int in_quote;
   char sep;
   int redir_handle = -1;
@@ -316,7 +316,7 @@ open_input_stream (char *cmd)
     {
       char *fileprefix;
 
-      fileprefix = choose_temp_base ();
+      fileprefix = make_temp_file (NULL);
       cpp_temp_file = (char *) xmalloc (strlen (fileprefix) + 5);
       sprintf (cpp_temp_file, "%s.irc", fileprefix);
       free (fileprefix);
-- 
2.35.3

